### DISCUSSION TITLE
How to make first major release?

### DISCUSSION THREAD

**Discussion by zenflow** [semantic-release/semantic-release#1916] [Q&A] [answered]
So you're doing initial development of a package ([major version zero](https://semver.org/#spec-item-4)), adding features and fixing bugs and whatnot (including possible breaking changes), until eventually you're happy with it and are ready to publish v1.0.0.

How should one go about publishing (or triggering the publishing of) v1.0.0 ?
Is there a recommended way to do this?

I can see two options, but maybe there is a better one:
1. Trigger the release by pushing an empty commit with `BREAKING CHANGE` in the commit message. This doesn't really seem appropriate though, when there really was no breaking change.
2. Make the release manually (with all the steps like bumping the package's `version` field, adding to changelog, making the commit, tagging the commit, etc)

Anyone have any insight to offer here?

**Comment by gr2m** [comment, marked as answer]
I wouldn't publish any 0.x releases. I'd instead use a `beta` release channel to release `1.0.0-beta.x` until you are happy with the result. That way you don't need any `BREAKING CHANGE:` in your commits

**Reply by zenflow** [reply]
@gr2m Awesome! Thanks for the answer! Good to know for next time I start a package /w semantic-release
